Assets/Scripts/Map/Partition.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Utilities;
+
+using UnityEngine;
+
+
+namespace Maes.Map
+{
+    public class Partition
+    {
+        public List<Vertex> Vertices { get; }
+        public Bitmap CommunicationZone { get; }
+        public int PartitionId { get; }
+        public Dictionary<(int, int), float> CommunicationRatio { get; }
+        public Dictionary<(int, int), Bitmap> IntersectionZones { get; }
+        public Dictionary<Vector2Int, Bitmap> CommunicationZones { get; }
+
+        public Partition(int partitionId, List<Vertex> vertices, Dictionary<Vector2Int, Bitmap> communicationZones)
+        {
+            PartitionId = partitionId;
+            Vertices = vertices;
+            CommunicationZones = communicationZones;
+            CommunicationZone = CreatePartitionCommunicationZone(communicationZones);
+            IntersectionZones = new Dictionary<(int, int), Bitmap>();
+            CommunicationRatio = new Dictionary<(int, int), float>();
+        }
+
+        public void CalculateIntersectionAndRatio(Partition otherPartition)
+        {
+            var key = (PartitionId, otherPartition.PartitionId);
+
+            if (AlreadyCalculated(key))
+            {
+                return;
+            }
```

Comment:
The code should probably take care of this. Are there any scenario where CalculateIntersectionAndRatio is called with the same partition more than once?

Reply:
Det er da muligt at kalde funktionen flere gange, dette gøre bare at vi ikke kalder unødvendig complexity igen

Reply:
Det er ikke denne funktions problem, det er dem der kalder funktionens problem